### DISCUSSION TITLE
M365 Search bar, people suggestions and pnp search results page

### DISCUSSION THREAD

**Discussion by bernhardk4t** [microsoft-search/pnp-modern-search#3082] [Q&A] [answered]
We have modified our SharePoint intranet, so that the search in the M365 on the top of the page is opening an individual search result page with pnp-modern-search webparts (including filters...).

When a search is entered, e.g. "test", the URL to the page looks like this:
https://demo.sharepoint.com/sites/intranet/SitePages/searchresults.aspx?q=test&k=test

Now query suggestions are enabled, and in the dropdown while typing in the search word there are also people displayed. When one clicks on one of the person suggestions, the following URL is opened (IDs modified):
https://demo.sharepoint.com/sites/intranet/SitePages/searchresults.aspx?q=&k=&pp=234abd56-79cc-4ea2-93a8-126bd24d2779@25d1125c-9710-4fd2-8d99-dde6cc5617ee|test.user@demo.de

Now with this URL the search result webpart fails. also if I would use the whole string **234abd56-79cc-4ea2-93a8-126bd24d2779@25d1125c-9710-4fd2-8d99-dde6cc5617ee|test.user@demo.de** in search it will not show any result.

Does anyone have an idea, how to handle the people suggestions in the search box with the pnp-modern-search webparts?

Best regards, Bernhard

**Comment by till-llit** [comment]
Hi Bernhard,

we had the same issue here and found no out of the box solution with PNP Modern Search - that's why we created a redirect webpart.

the webpart is able to listen to url parameters (like pp=) in this case and use a regex to split the value (to get the email) and in a last step redirect the user to the Delve (or any other people profile display page) with the email as parameter.

im not allowed to share the webpart, but just to let you know that there is a way to handle this use case.

**Reply by bernhardk4t** [reply]
Thanks. I will give it a try if this can also be done using the React Script Editor Webpart (https://github.com/pnp/sp-dev-fx-webparts/tree/main/samples/react-script-editor).

**Comment by wobba** [comment, marked as answer]
Couple of options.

* Change the scope of the search box via PowerShell to be Site and not Tenant as this removes people suggestions - and you can set the placeholder text as well to match you scenario
* Hide the header search box using PowerShell, and use the PnP search box on the page instead.

**Reply by bernhardk4t** [reply]
Yup, that scope is the solution for our scenario - hadn't thought of this setting... Thanks a lot!
Bernhard
